contracts/vortex/CarbonVortex.sol

```diff
@@ -0,0 +1,262 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+import { ICarbonVortex } from "./interfaces/ICarbonVortex.sol";
+import { IVersioned } from "../utility/interfaces/IVersioned.sol";
+import { ICarbonController } from "../carbon/interfaces/ICarbonController.sol";
+import { Upgradeable } from "../utility/Upgradeable.sol";
+import { Token } from "../token/Token.sol";
+import { Utils } from "../utility/Utils.sol";
+import { MathEx } from "../utility/MathEx.sol";
+import { MAX_GAP, PPM_RESOLUTION } from "../utility/Constants.sol";
+
+interface IBancorNetwork {
+    function collectionByPool(Token pool) external view returns (address);
+
+    function tradeBySourceAmount(
+        Token sourceToken,
+        Token targetToken,
+        uint256 sourceAmount,
+        uint256 minReturnAmount,
+        uint256 deadline,
+        address beneficiary
+    ) external payable returns (uint256);
+}
+
+/**
+ * @dev CarbonVortex contract
+ */
+contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable, Utils {
+    using Address for address payable;
+
+    ICarbonController private immutable _carbonController;
+    IBancorNetwork private immutable _bancorNetwork;
+    Token private immutable _bnt;
+
+    uint256 private _totalBurnt;
+
+    // rewards percentage
+    uint256 private _rewardsPercentagePPM;
```

Comment:
`uint32 private _rewardsPPM;`

Reply:
updated (keep in mind it should be uint256 so as not to corrupt the next storage slot during an upgrade)

Reply:
It's a new contract (first-time deployment, not an upgrade); how can any storage be corrupted here?

Reply:
it will be upgraded from `FeeBurner`, the name is changed

Reply:
So this:
```
uint256 private _totalBurnt;
uint32 percentagePPM;
uint224 maxAmount;
```
Changes to this:
```
uint256 private _totalBurned;
uint256 private _rewardsPPM;
```
Which makes it fine to declare `_rewardsPPM` as `uint32`.

Reply:
It's not fine - if we introduce another variable, it may be packed in the same slot (if it's <= uint224), which will be corrupted with data from maxAmount.